### PR TITLE
Added security checks for str.format. [5.0]

### PR DESCRIPTION
**Pull request by mauritsvanrees**
Part of PloneHotfix20170117.

### CODE REVIEW

**Comment by mauritsvanrees** [comment]
@vangheem This merges the hotfix into 5.0.

**Comment by mauritsvanrees** [comment]
Note that this adds the formatting function and classes from the hotfix into `utils.py`. So third party code might start relying on this. If we do not want this, then we had better change that now, before merging.

**Comment by jensens** [comment]
Well, utils.py is probably the right location. if we want to indicate this a thing not to use outside we could prefix all with ``_`` - but I am not sure if this is needed/ what we want.

**Comment by jensens** [comment]
Since nobody has an opposite opinion on this I will merge it.

File: Products/CMFPlone/resources/browser/mixins.py [hunk 1]
```diff
@@ -67,9 +68,10 @@ def __call__(self):
             less_vars_params[name] = value
 
         for name, value in registry.items():
-            t = value.format(**less_vars_params)
+            t = SafeFormatter(value).safe_format(**less_vars_params)
```
Comment (thet) [inline Products/CMFPlone/resources/browser/mixins.py]: @mauritsvanrees you have imported ``safe_format`` but using ``SafeFormatter``. This leads to an error mentioned here: https://github.com/plone/Products.CMFPlone/pull/1938#issuecomment-277922063

File: Products/CMFPlone/resources/browser/mixins.py [hunk 2]
```diff
@@ -115,7 +117,7 @@ def __call__(self):
...
```
Comment (thet) [inline Products/CMFPlone/resources/browser/mixins.py]: Same here.

**Comment by mauritsvanrees** [comment]
Huh, strange that this is not caught by the tests.
I have added pull request https://github.com/plone/Products.CMFPlone/pull/1944 which fixes the import and hopefully makes things a bit clearer.

**Pull request merged**